### ISSUE TITLE
Aggregator refresh.

### ISSUE THREAD

**Issue by ShreeSam** [tfsaggregator/tfsaggregator#175]
Hi Team,

I have a custom integer field called PO Estimate that has roll up aggregation of all child PO Estimate.
now i have run into an issue with the roll-up of the PO Estimate in the Purchasing 2.1 release epics.

I created epic PO-19 which originally included tasks that I moved to epic PO-21. I changed the parent link on the user stories from PO-19 to PO-21. However, the PO Estimate hours for those tasks is not being removed from the PO-19 roll-up.

The total PO estimate on epic PO-19 is showing 804 hours. It should be 740 hours as 64 hours were moved from PO-19 to PO-21
Also - is there any way to auto-update the rollups? Right now, I have to open each feature and epic and refresh it to be sure that the numbers I'm looking at are the correct rollup. We really need these values to be auto-updated so when I display the data in TFS the hours are accurate. If we have to refresh each feature and epic to get the right hours, that is going to be a waste of time.  

Thanks,
Shree

**Comment by jessehouwing** [comment]
Is the rollup configured to act on changes of the parent,a child or both?

**Comment by ShreeSam** [comment]
Rollup configured to act on changes of the child

**Comment by ShreeSam** [comment]
Rollup configure to act on child.

**Comment by jessehouwing** [comment]
You may need a rule defined on the parent level as well, depending on where
you changed the link between these work items, the rule may trigger as a
change on the parent and not in the child.

**Comment by ShreeSam** [comment]
Ok. Will change configuration.
I have one more query.

Is there any way to auto-update the rollups? Right now, I have to open each feature and epic and refresh it to be sure that the numbers I'm looking at are the correct rollup. We really need these values to be auto-updated so when I display the data in TFS the hours are accurate. If we have to refresh each feature and epic to get the right hours, that is going to be a waste of time.

Thanks,
Shree

**Comment by giuliov** [comment]
For a one-shot operation, you can use the command line tool: it accepts a list of work items ids. To get the list of interested ids, you may win a query and get a CSV file.

----- Messaggio originale -----
Da: "Shreedevi Bhat" <notifications@github.com>
Inviato: ‎30/‎08/‎2016 13:11
A: "tfsaggregator/tfsaggregator" <tfsaggregator@noreply.github.com>
Oggetto: Re: [tfsaggregator/tfsaggregator] Aggregator refresh. (#175)

Ok. Will change configuration.
I have one more query.
Is there any way to auto-update the rollups? Right now, I have to open each feature and epic and refresh it to be sure that the numbers I'm looking at are the correct rollup. We really need these values to be auto-updated so when I display the data in TFS the hours are accurate. If we have to refresh each feature and epic to get the right hours, that is going to be a waste of time.
Thanks,
Shree

**Comment by ShreeSam** [comment]
Hi,

Am trying to run command line tool for a list of work items but it says " Input string was not in a correct format"
Below is my command line arguments.
TFSAggregator2.ConsoleApp.exe run --policyFile="C:\Program Files\Microsoft Team Foundation Server 14.0\Application Tier\Web Services\bin\Plugins\T FSAggregator2.ServerPlugin.policies" --teamProjectCollectionUrl=http://<serverName>:8080/tfs/TPC --teamProjectName="Team Project" --workItemId="16047, 16346"

i tried 
TFSAggregator2.ConsoleApp.exe run --policyFile="C:\Program Files\Microsoft Team Foundation Server 14.0\Application Tier\Web Services\bin\Plugins\T FSAggregator2.ServerPlugin.policies" --teamProjectCollectionUrl=http://<serverName>:8080/tfs/TPC --teamProjectName="Team Project" --workItemId=16047, 16346

Please help me with this.

Thanks in Advance,
Shree

**Comment by ShreeSam** [comment]
Hi ,

I have define rollup for both to child and parent. even though below scenario is not working.

I have a custom integer field called PO Estimate that has roll up aggregation of all child PO Estimate.
now i have run into an issue with the roll-up of the PO Estimate in the Purchasing 2.1 release epics.

I created epic PO-19 which originally included tasks that I moved to epic PO-21. I changed the parent link on the user stories from PO-19 to PO-21. However, the PO Estimate hours for those tasks is not being removed from the PO-19 roll-up.

The total PO estimate on epic PO-19 is showing 804 hours. It should be 740 hours as 64 hours were moved from PO-19 to PO-21

Thanks,
Shree
